### PR TITLE
Add package to test whether value is a startcase string

### LINKED ISSUE #538: [RFC]: add support for testing whether a value is start-cased string
### Description

This RFC proposes to add support for testing whether a value is a start-cased string.

```javascript
var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
var startcase = require( '@stdlib/string/base/startcase' );

function isStartcase( value ) {
    return ( isString( value ) && startcase( value ) === value );
}
```

Package: `@stdlib/assert/is-startcase`
alias: `isStartcase`

### Related Issues

No.

### Questions

No.

### Other

Reference packages:

-   [`@stdlib/assert/is-uppercase`](https://github.com/stdlib-js/stdlib/tree/13c1fa39ff73208d819697dc07917bdb7539f295/lib/node_modules/%40stdlib/assert/is-uppercase)

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Planeshifter**
Resolves #538.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a package to test whether value is a startcase string

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #538
-   fixes #

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Planeshifter** [comment]
@kgryte Package is ready for review.

File: lib/node_modules/@stdlib/assert/is-startcase/README.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+## Usage
+
+```javascript
+var isStartcase = require( '@stdlib/assert/is-startcase' );
+```
+
+#### isStartcase( value )
+
+Tests if a `value` is a startcase `string`, where the first character of each word is uppercase.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/README.md]:
```suggestion
Tests if a `value` is a startcase `string` (i.e., the first character of each word is uppercase).
```

File: lib/node_modules/@stdlib/assert/is-startcase/benchmark/benchmark.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		'BeepANDBoop',
+		'BEEP_BOOP',
+		null,
+		void 0,
+		NaN,
+		true,
+		false,
+		[],
+		{},
+		function noop() {}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/benchmark/benchmark.js]:
```suggestion
		{}
```

File: lib/node_modules/@stdlib/assert/is-startcase/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+
+{{alias}}( value )
+    Tests if a value is a startcase string, where the first character of each
+    word is uppercase.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/docs/repl.txt]:
```suggestion
    Tests if a value is a startcase string.
```

File: lib/node_modules/@stdlib/assert/is-startcase/docs/types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 2.0
+
+/**
+* Tests if a value is a startcase string, where the first character of each word is uppercase.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/docs/types/index.d.ts]:
```suggestion
* Tests if a value is a startcase string (i.e., the first character of each word is uppercase).
```

File: lib/node_modules/@stdlib/assert/is-startcase/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+*
+* @param value - value to test
+* @returns boolean indicating if value is a startcase string
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/docs/types/index.d.ts]:
```suggestion
* @returns boolean indicating if a value is a startcase string
```

File: lib/node_modules/@stdlib/assert/is-startcase/lib/main.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+var startcase = require( '@stdlib/string/base/startcase' );
+
+
+// MAIN //
+
+/**
+* Tests if a value is a startcase string, where the first character of each word is uppercase.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/lib/main.js]:
```suggestion
* Tests if a value is a startcase string (i.e., the first character of each word is uppercase).
```

File: lib/node_modules/@stdlib/assert/is-startcase/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+*
+* @param {*} value - value to test
+* @returns {boolean} boolean indicating if value is a startcase string
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/lib/main.js]:
```suggestion
* @returns {boolean} boolean indicating if a value is a startcase string
```

File: lib/node_modules/@stdlib/assert/is-startcase/test/test.cli.js [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+			t.fail( error.message );
+		} else {
+			t.strictEqual( stdout.toString(), '', 'does not print to `stdout`' );
+			t.strictEqual( stderr.toString(), PKG_VERSION+'\n', 'expected value' );
+		}
+		t.end();
+	}
+});
+
+tape( 'the command-line interface tests if a string argument matches a startcase value', opts, function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-startcase/test/test.cli.js]:
```suggestion
tape( 'the command-line interface tests if a value is a startcase string', opts, function test( t ) {
```

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
LGTM. Will merge once all checks pass.

**Comment by kgryte** [comment]
Linting fails, but this seems due to files which have not been added in this PR. Will merge...

**Pull request merged**